clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/OperationsController.groovy

```diff
@@ -49,165 +31,53 @@ import org.springframework.web.bind.annotation.RestController
 @RestController
 class OperationsController {
 
-  @Autowired MessageSource messageSource
-  @Autowired OrchestrationProcessor orchestrationProcessor
-  @Autowired Registry registry
-  @Autowired (required = false) Collection<AllowedAccountsValidator> allowedAccountValidators = []
-  @Autowired (required = false) List<AtomicOperationDescriptionPreProcessor> atomicOperationDescriptionPreProcessors = []
-  @Autowired AtomicOperationsRegistry atomicOperationsRegistry
-  @Autowired SecurityConfig.OperationsSecurityConfigurationProperties opsSecurityConfigProps
-  @Autowired AccountCredentialsRepository accountCredentialsRepository
+  private final OperationsService operationsService
 
-  @Autowired DescriptionAuthorizer descriptionAuthorizer
-
-  /*
-   * APIs
-   * ----------------------------------------------------------------------------------------------------------------------------
-   */
-
-  /**
+  @Autowired
+  OperationsController(OperationsService operationsService) {
+    this.operationsService = operationsService
+  }
+/**
    * @deprecated Use /{cloudProvider}/ops instead
    */
   @Deprecated
   @RequestMapping(value = "/ops", method = RequestMethod.POST)
-  Map<String, String> operations(@RequestParam(value = "clientRequestId", required = false) String clientRequestId,
-                                 @RequestBody List<Map<String, Map>> requestBody) {
-    List<AtomicOperation> atomicOperations = collectAtomicOperations(requestBody)
-    start(atomicOperations, clientRequestId)
+  OperationsService.StartOperationResult operations(
+    @RequestParam(value = "clientRequestId", required = false) String clientRequestId,
+    @RequestBody List<Map<String, Map>> requestBody) {
+    List<AtomicOperation> atomicOperations = operationsService.collectAtomicOperations(requestBody)
```

Comment:
Calling one method of `operationsService` to get a value (`atomicOperations`), just to pass that value to another method of the same service, is a bad smell. The pattern is repeated several times in this class. 

Can both calls be encapsulated into one method? Or is that an indication that `collectAtomicOperations` and `start` should belong to separate services?

Reply:
Good point. Thank you - I'll split these into two services. I wasn't too pleased with this myself, thanks for keeping me honest. :)